File: Binary Search/Java/BinarySearch.java
class BinarySearch {
    int binarySearch(int arr[], int l, int r, int data) {
        if (r >= l) {
            int mid = (l + r) / 2;
            if (arr[mid] == data)
                return mid;
            if (arr[mid] > data)
                return binarySearch(arr, l, mid - 1, data);
            return binarySearch(arr, mid + 1, r, data);
        }
        return -1;
    }

    public static void main(String args[]) {
        BinarySearch bs = new BinarySearch();
        int arr[] = { 6, 8, 12, 16, 21, 34, 51 };
        int n = arr.length;
        int data = 21;
        int result = bs.binarySearch(arr, 0, n - 1, data);
        if (result == -1)
            System.out.println("Element is not found");
        else
            System.out.println("Element found at index " + result);
    }
}

File: Binary Search/Python/binary_search.py

def binarySearch(arr, l, r, data):
    if r >= l:
        mid = (l + r) // 2
        if arr[mid] == data:
            return mid

        elif arr[mid] > data:
            return binarySearch(arr, l, mid-1, data)

        else:
            return binarySearch(arr, mid + 1, r, data)

    else:
        return -1


arr = [6, 8, 12, 16, 21, 34, 51]
data = 18
result = binarySearch(arr, 0, len(arr)-1, data)

if(result == -1):
    print("Element not found")
else:
    print("Element found at index", result)
